src/manager/video.manager.ts

```diff
@@ -69,4 +72,14 @@ export class VideoManager {
     public getVideo(id: string): Promise<Video> {
         return this.videoDatastore.getById(id);
     }
+
+    public async updateVideo(id: string, item: Video): Promise<Video> {
+        const oldVideo = await this.mediaDatastore.getById(id);
+        const newVideo = await this.videoDatastore.getOneByOptionsAndUpdate({ _id: id }, item);
+
+        this.talentDatastore.removeMediaById({ _id: oldVideo.talent }, id);
+        this.talentDatastore.addMediaToTalent({ _id: newVideo.talent }, newVideo);
+
+        return newVideo;
```

Comment:
You don't have to wait so much. Use a promise.all to do them all in the same time and return the result[index] of the promise you actually want to return